<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
<head>
  <meta charset="utf-8" />
  <meta name="generator" content="pandoc" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
  <title>Dynamic and thermodynamic models of adaptation</title>
  <link rel="stylesheet" href="https://latex.now.sh/style.min.css" />
  <link rel="stylesheet" href="https://unpkg.com/latex.css/style.min.css" />
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    span.underline{text-decoration: underline;}
    div.column{display: inline-block; vertical-align: top; width: 50%;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    .display.math{display: block; text-align: center; margin: 0.5rem auto;}
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
    }
    .hanging div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }
  </style>
  <link rel="stylesheet" href="https://latex.now.sh/style.min.css" />
  <link rel="stylesheet" href="https://unpkg.com/latex.css/style.min.css" />
  <!-- <link rel="stylesheet" href="/home/guest/Документы/aysuvorov.github.io/assets/css/style.min.css" /> -->

  <!--[if lt IE 9]>
    <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv-printshiv.min.js"></script>
  <![endif]-->
</head>
<body>
<header id="title-block-header">
<h1 class="title">Dynamic and thermodynamic models of adaptation</h1>
</header>
<script type="text/javascript" id="MathJax-script" async
  src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js">
</script>
<script>
MathJax = {
  tex: {
    inlineMath: [['$', '$'], ['\\(', '\\)']]
  }
};
</script>
<script id="MathJax-script" async
  src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml.js">
</script>
<h2 id="dynamic-and-thermodynamic-models-of-adaptation">Dynamic and thermodynamic models of adaptation</h2>
<p align="center">
<img src="https://i.pinimg.com/564x/85/0e/fa/850efa7658673ea37edfdbece9e44641.jpg" />
</p>
<p>Overview of works <a href="https://orcid.org/0000-0002-5839-0393">M. Zanin</a>, <a href="https://orcid.org/0000-0001-6224-1430">A. Gorban</a>, <a href="https://orcid.org/0000-0001-7540-1130">A. Zaikin</a>, <a href="https://orcid.org/0000-0001-8987-4158">H. Whitwell</a>, <a href="https://orcid.org/0000-0002-1169-5149">M. Krivonisov</a>, <a href="https://orcid.org/0000-0002-4245-7346">T. Nazarenko</a></p>
<p>by Aleksandr Suvorov</p>
<p>5.01.2023</p>
<h2 id="abstract">Abstract</h2>
<p>Network analysis provides great opportunities for modelling complex biological systems. There are different approaches for modelling such systems with network and in this article we will consider some basic usefull approaches. The main idea is focused on building pipelines whrer individual units are represented as graphs. The construction of such pipelines can make it possible to solve problems of classification, regression, and analyzштп dynamic changes in biological systems.</p>
<p>We will discuss common approaches for different tasks. The chapter contains the instruction for building graphs step by step focusing on the strengths and weaknesses of a particular approach.</p>
<p>Network analysis can become one of the leading method for building explainable artificial intelligence (AI) models.</p>
<p>The method is highly perspective due to the available and wide visualization tools, as well as analytical instruments based on the topological properties of graphs.</p>
<h3 id="key-words">Key words</h3>
<p>Networks, network analysis, graphs, machine learning, artificial intelligence, complex models, correlation, statistical modelling, classification, regression</p>
<h2 id="introduction">Introduction</h2>
<p>Network theory made it possible to take a different look at biological systems and processes. Unlike simple models that describe specific phenomenons, network analysis allows to take into account complex interacting factors or features. Such features can be represented as graph nodes that will reflect a particular system (organism, cell, population, etc., that is, some experimental unit of interest).</p>
<p>Features somehow interact with each other and the nature of their interaction can be estimated in some way. Such an estimate can be represented as an attribute or weight of an edge between feature nodes.</p>
<p>Of course, in biological systems the features of the ongoing processes are extremely variable and may have complex distributions. Thus, it seems extremely difficult to determine some models that describe the interactions between features and estimate the weight of the factors-nodes themselves. Such models must be extremely complex and non-linear. The choice of specific models can be done empirically.</p>
<p>Moreover, network analysis also allows us to evaluate system dynamic changes in the structure of the graph, which brings this type of analysis even closer to a realistic description of biological systems. When assessing dynamic fluctuations in such structures, promising areas are the assessment of system reactions to any external stimuli, the assessment of adaptation, “crisis” situations and complete failure of adaptation.</p>
<h2 id="materials">Materials</h2>
<p>There is a great number of different models and approaches that allow one or another biological experimental unit to be represented as a network of basic features. We will focus on some of them in order to provide a basic understanding of how interactions between factors are formed and evaluated, which will allow the use of a network analysis approach in descriptive and predictive models.</p>
<h3 id="biological-system-model">Biological system model</h3>
<p>The most important question is of choosing the most representative descriptive model of relations in a graph. The graph below shows a weighted graph <span class="math inline"><em>G</em></span> with <span class="math inline"><em>V</em></span> nodes and <span class="math inline"><em>E</em></span> edges.</p>
<p>Nodes represent features of the experimental unit. They can be ranked, and then their weight will be different (on the chart, nodes with different weights have different sizes). The edges weights characterize a certain assessment of the relations between the nodes. The thickness of the edges on the graph is proportional to the weights. The connection may be completely absent (on the graph - nodes A and C, A and G), then the edge is not built.</p>
<p><img src="./images/Figure-1-weighted-graph.png" alt="Primer of weighted graph" /> <strong>Figure 1.</strong> An example of a weighted graph. Vertices and edges have a size and thickness based on their weights.</p>
<p>Theoretically, the graph can also be directed, in certain situations this can increase the realism of the model.</p>
<p>The fundamental issue is the choice of the system that we consider as a network.</p>
<p>Such systems can be:</p>
<ol type="1">
<li><p>Some individual experimental units of one set (patients, bacterial colonies, laboratory animals, protein systems, etc.) - in this case, the graph will represent such individual unit. All other units should have a similar set of features, while differences will be noted only in the weights of the edges (representing differencies in relation between unit features);</p></li>
<li><p>One single system with properties dynamically changing over time. In this case, its states at different time points can be represented as graphs. Such networks should also have a similar set of features (nodes) with interconnections (edge weights) changing over time;</p></li>
<li><p>Some individual experimental units of one set, the properties of which change over time. Thus, a set of graphs will be obtained that characterizes each individual experimental unit, as in case (1), and also for each such unit it will be possible to construct dynamic graphs, as in case (2).</p></li>
</ol>
<h2 id="methods">Methods</h2>
<p>There are several steps in building graphs for experimental units:</p>
<ol type="1">
<li><p>The choice of the systems that we want to represent as a network and the tasks we want to solve - will it be a classification task or an assessment of dynamic changes, etc.? This step precedes the selection of an appropriate model to calculate the attributes of the edges. Thus, the analysis of dynamic changes in the system indicates the choice of a correlation model. The task of classification is rather associated with the choice of other types of models;</p></li>
<li><p>Preliminary data transformation;</p></li>
<li><p>Determining the features that will be the nodes of the future graphs;</p></li>
<li><p>Choice of a model and a method that will allow estimating the weights of the edges;</p></li>
<li><p>Transforming units into graphs;</p></li>
<li><p>Next, we can go one of two ways:</p>
<ul>
<li><p>Calculation specific graph characterics and subsequent calculations with these characterics, as with new features of the experimental unit. We can then transform the data and send it to be trained by some classifier (machine learning algorithm, neural network, etc.);</p></li>
<li><p>Sending a whole graph or a list of graphs for training in a graph neural network (GNN).</p></li>
</ul></li>
</ol>
<h3 id="assigning-weights-to-edges">Assigning weights to edges</h3>
<p>The vast majority of the features at the level of biochemical parameters, immunological reactions, genetic polymorphisms, etc. are quantitative. When considering the other levels of networks, e.g. patients as units, their medical history and examination results, most of them are qualitative.</p>
<p>The calculation of the edge weight is usually based on some model that evaluates the relationship between the characteristics. The simplest option implies that a single type of model can be used to evaluate all pairwise relationships between features. Various model results and indicators can be used as a relation measure: it can be the magnitude of the error in the linear regression <span class="citation" data-cites="Zanin2011Sep">[1]</span>, <span class="citation" data-cites="Nazarenko2021Oct">[2]</span>, the calculated probability of belonging to the class<span class="citation" data-cites="Krivonosov2022Jan">[3]</span>, the correlation<span class="citation" data-cites="Gorban2009May">[4]</span>, or something else. Thus, it is obvious that the type of model used to calculate the weights should be as universal as possible and describe various types of relationships, including non-linear ones.</p>
<p>There are three popular approaches for assigning weight to an edge:</p>
<ul>
<li>direct assignment of weight to the edge, as a result a weighted graph is constructed;</li>
<li>construction of an edge, provided that the calculated model indicator is above or below a certain threshold. The edge is built, and the weight value corresponds to 1. In this case it is necessary to set a certain threshold value. The threshold may be known, but it may also be determined empirically;</li>
<li>a combination of both previous approaches: the edge is built under the condition that the calculated indicator is above/below the threshold, while the weight of the edge corresponds to the calculated model indicator. If the value of the indicator does not exceed the threshold, the edge is absent.</li>
</ul>
<p>Based on these approaches various methods are mentioned in the literature:</p>
<ul>
<li>parenclitic graphs using threshold<span class="citation" data-cites="Zanin2011Sep">[1]</span>;</li>
<li>weighted parenclitic graphs<span class="citation" data-cites="Nazarenko2021Oct">[2]</span>, <span class="citation" data-cites="Whitwell2018Apr">[5]</span>;</li>
<li>weighted sinolithic graphs<span class="citation" data-cites="Krivonosov2022Jan">[3]</span>;</li>
<li>correlation graphs using threshold<span class="citation" data-cites="Gorban2009May">[4]</span>;</li>
<li>weighted correlation graphs<span class="citation" data-cites="Gorban2009May">[4]</span>.</li>
</ul>
<p>Let us consider in detail the process of constructing various graphs of each type.</p>
<h3 id="basic-graph-building-methods">Basic graph building methods</h3>
<h4 id="parenclitic-networks-using-linear-model-and-threshold-tlrpa">Parenclitic networks using linear model and threshold (tLRPA)</h4>
<p>The parenclitic approach bases on the hypothesis that there is a linear relation between the features of “normal” or “healthy” units. The process of calculating the edge weights is based on a linear regression equation. Let’s take a closer look at the step by step process.</p>
<ol type="1">
<li><p>The set of “healthy” or “normal” units is defined (control group).</p></li>
<li><p>The features that will be the future vertices of the graph are defined, for example, {A, B, C}. Pairwise combinations without repetitions between the features of the unit are determined: {A-B; A-C; B-C}. For further construction of graphs it is important that the combinations themselves and the order of elements are identical for all experimental units.</p></li>
<li><p>For each pairwise combination of characteristics (<span class="math inline"><em>x</em></span>, <span class="math inline"><em>y</em></span>) of the entire control group, a linear regression model is fitted:</p>
<p><span class="math display"><em>Y</em> = <em>X</em><em>β</em><sub><em>x</em><em>y</em></sub> + <em>ϵ</em></span></p>
<p>, where <span class="math inline"><em>ϵ</em></span> will be an array with residuals.</p>
<p>Thus, for each pair of characteristics (<span class="math inline"><em>x</em></span>, <span class="math inline"><em>y</em></span>) for <span class="math inline"><em>n</em></span> experimental units, there is an array with residuals [<span class="math inline"><em>ϵ</em><sub>1</sub>, <em>ϵ</em><sub>2</sub>, <em>ϵ</em><sub>3</sub>, ... <em>ϵ</em><sub><em>n</em></sub></span>]. Since we know the value of both features, we can calculate <span class="math inline"><em>ϵ</em><sub><em>i</em></sub></span> as follows:</p>
<p><span class="math display"><em>ϵ</em><sub><em>i</em></sub> = <em>β</em><sub>0</sub> + <em>β</em><sub>1</sub><em>x</em><sub><em>i</em></sub> − <em>y</em><sub><em>i</em></sub></span></p></li>
<li><p>For the control group, for each pair of characteristics (<span class="math inline"><em>x</em></span>, <span class="math inline"><em>y</em></span>), linear regression coefficients <span class="math inline"><em>β</em><sub>0<em>x</em><em>y</em></sub>, <em>β</em><sub>1<em>x</em><em>y</em></sub></span> are also obtained.</p></li>
<li><p>Array with residuals is converted to absolute Z-values, |Z-scores|:</p>
<p><span class="math display">$$
  \begin{equation}
  abs(z_i) = \frac{|\epsilon_i - \hat{\epsilon}_{CTRL}|}{\sigma(\epsilon_{CTRL})}
  \end{equation}
  $$</span></p>
<p>, where</p>
<ul>
<li><span class="math inline"><em>i</em></span> - experimental unit,</li>
<li><span class="math inline"><em>ϵ̂</em><sub><em>C</em><em>T</em><em>R</em><em>L</em></sub></span> - mean residual <span class="math inline"><em>ϵ</em><sub>1 : <em>i</em></sub></span> from the linear regression equation for features (<span class="math inline"><em>x</em></span>, <span class="math inline"><em>y</em></span>) for the control group,</li>
<li><span class="math inline"><em>σ</em>(<em>ϵ</em><sub><em>C</em><em>T</em><em>R</em><em>L</em></sub>)</span> - standard deviation of <span class="math inline"><em>ϵ</em><sub>1 : <em>i</em></sub></span> residual values for the control group.</li>
</ul></li>
<li><p>For each pair of characteristics (<span class="math inline"><em>x</em></span>, <span class="math inline"><em>y</em></span>) we have obtained linear regression coefficients (<span class="math inline"><em>β</em><sub>0<em>x</em><em>y</em></sub>, <em>β</em><sub>1<em>x</em><em>y</em></sub></span>). All intermediate stages before graph construction are completed.</p></li>
<li><p>Next, we get some experimental unit <span class="math inline"><em>i</em></span> that is of interest to us. We substitute the values of the features (<span class="math inline"><em>x</em><sub><em>i</em></sub></span>, <span class="math inline"><em>y</em><sub><em>i</em></sub></span>) into the already obtained linear regression equations and obtain its own value <span class="math inline"><em>ϵ</em><sub><em>i</em></sub></span>. Further, we calculate absolute Z-score from formula (3). We repeat the prossess for every new experimental unit.</p></li>
</ol>
<p><img src="./images/Figure-2-lm-regr-graph.png" alt="Figure 2" /> <strong>Figure 2.</strong> Linear regression between features (<span class="math inline"><em>x</em></span>, <span class="math inline"><em>y</em></span>). Each blue dot represents an experimental unit. The regression residuals are green or red. The gray lines show the boundaries of the threshold. If the normalized absolute values of the residuals do not exceed the threshold, they are colored green. When the threshold is exceeded, the residuals are colored red.</p>
<ol type="1">
<li>From each experimental unit, a graph <span class="math inline"><em>G</em>(<em>V</em>,<em>E</em>)</span> is constructed, the features become the set of nodes <span class="math inline"><em>V</em></span>, and <span class="math inline"><em>E</em></span> - the set of edges. If the calculated value <span class="math inline"><em>a</em><em>b</em><em>s</em>(<em>z</em><sub><em>i</em></sub>)</span> for a pair of nodes (<span class="math inline"><em>x</em></span>, <span class="math inline"><em>y</em></span>) is greater than or equal to a certain threshold, an edge is built. If the value is less than the threshold, there is no edge. Thus, units similar to the control group should have small <span class="math inline"><em>a</em><em>b</em><em>s</em>(<em>z</em><sub><em>i</em></sub>)</span> and graphs representing theese units will have few edges. Units highly deviating from control group will have large <span class="math inline"><em>a</em><em>b</em><em>s</em>(<em>z</em><sub><em>i</em></sub>)</span> and more constructed edges: if the residual is very large, the unit differs from the control group and does not obey the calculated linear relationship, and the edge will be built.</li>
</ol>
<p>The main challenge is in the determining the threshold. In the work of M.Zanin and S.Bocaletti <span class="citation" data-cites="Zanin2011Sep">[1]</span> the threshold was taken as <span class="math inline"><em>a</em><em>b</em><em>s</em>(<em>z</em>)</span> = 2.</p>
<p><img src="./images/Figure-3-threshold-graph.png" alt="Figure 3" /> <strong>Figure 3.</strong> An example of a graph with threshold values for a hypothetical experimental unit of a parenclitic network. Edges corresponding to <span class="math inline"><em>a</em><em>b</em><em>s</em>(<em>z</em>)</span> below threshold 2 are not built (sharp red dotted line). Edges that correspond to <span class="math inline"><em>a</em><em>b</em><em>s</em>(<em>z</em>)</span> greater than or equal to 2 are built (bright blue). The indicated numbers do not correspond to the weight of the edge, just to the calculated <span class="math inline"><em>a</em><em>b</em><em>s</em>(<em>z</em>)</span>.</p>
<h4 id="weighted-parenclitic-networks-using-linear-model-wlrpa">Weighted parenclitic networks using linear model (wLRPA)</h4>
<p>The need of the threshold raises large number of challenges. One way to get rid of the threshold is to build a weighted graph. The step by step process is as follows:</p>
<ol type="1">
<li><p>Follow the steps similar to 1. - 7. for the parenclitic approach with a threshold.</p></li>
<li><p>Since there is no threshold, all edges of the graph are constructed, but each edge is assigned a weight equal to the corresponding value of the absolute deviation <span class="math inline"><em>a</em><em>b</em><em>s</em>(<em>z</em>)</span>. The resulting graphs are weighted.</p></li>
</ol>
<p><img src="./images/Figure-4-weighted-full-graph.png" alt="Figure 4" /> <strong>Figure 4.</strong> An example of a weighted graph of a hypothetical experimental unit of a parenclitic network. The weight of the edge and the edge thickness corresponds to <span class="math inline"><em>a</em><em>b</em><em>s</em>(<em>z</em>)</span>.</p>
<h4 id="parenclitic-networks-using-nuclear-density-estimation-wkdepa">Parenclitic networks using nuclear density estimation (wKDEPA)</h4>
<p>When using linear regression method for the construction of parenclitic graphs, we assumed that there is a linear relation between the features. However, this does not always happen, and the distribution of features is usually non-linear. Various approaches were proposed to describe both linear and non-linear relations in a single method. One of such approaches is kernel density estimation. Theoretically the approach can be used for both quantitative and categorical variables. In practice, the estimation of association between variables of different types differs <span class="citation" data-cites="Whitwell2018Apr">[5]</span>. Let’s take a look at the individual steps of this method.</p>
<ol type="1">
<li><p>Again, control group of “normal” units is specified. Pairwise combinations of features are also produced.</p></li>
<li><p>For each pairwise combination of features (<span class="math inline"><em>x</em></span>, <span class="math inline"><em>y</em></span>) a square two dimensional kernel density matrix is created which is called the <em>contour matrix</em>. The number of rows and columns of the matrix must exceed the length of the vectors <span class="math inline"><em>x</em></span> and <span class="math inline"><em>y</em></span>, but is chosen empirically, taking into account the number of experimental units and computational power.</p></li>
<li><p>On the constructed contour matrix, the point with the highest density becomes the so-called center.</p></li>
</ol>
<p><img src="./images/Figure-5-kde-topo.png" alt="Figure 5" /> <strong>Figure 5.</strong> An example of a contour density matrix for features (<span class="math inline"><em>x</em></span>, <span class="math inline"><em>y</em></span>). Red dots show individual experimental units. The center or point with the highest density is represented by a green diamond.</p>
<ol start="4" type="1">
<li><p>Generally, for each experimental unit <span class="math inline"><em>i</em></span> with coordinates by features [<span class="math inline"><em>x</em><sub><em>i</em></sub></span>, <span class="math inline"><em>y</em><sub><em>i</em></sub></span>] on the contour matrix, we calculate the distance from the center of the matrix to the coordinate point. Euclidean or Mahalanobis distance are popular approaches. The distance can serve as a raw edge weight for this unit. More complex idea is to estimate the volume under the densities that are higher than the density where the point is within the distribution.</p></li>
<li><p>Different approaches for calculating edge weights depending on feature distributions are described in Whitwell, H. J. et al. <span class="citation" data-cites="Whitwell2018Apr">[5]</span> and are experimental. Work in this direction continues.</p></li>
<li><p>By analogy with parenclitic networks using a linear model, an approach with direct assignment of a calculated weight to an edge (weighted network approach) can be used, or an approach using threshold (threshold approach) can be used. In the latter case the threshold value must be established empirically.</p></li>
</ol>
<h4 id="synolytic-networks-sa">Synolytic networks (SA)</h4>
<p>The synolytic approach is based on constructing the boundary between classes with the help of some classifier. The boundary must provide best separation between “normal” and “unnormal” units. For every pair of unit features a classifier is fitted to build such boundary and to relate the unit to one class or to another. The approach uses units both from “normal” and “unnormal” sets. In literature support vector machine (SVM) with the radial basis function kernel <span class="citation" data-cites="Krivonosov2022Jan">[3]</span>, as well as logistic regression <span class="citation" data-cites="Nazarenko2021Oct">[2]</span>, have been considered. However any suitable classifier may be used.</p>
<p>The sequence of manipulations when using the sinolytic approach is as follows:</p>
<ol type="1">
<li><p>Experimental units belonging to “normal” and “abnormal” classes are used. Pairwise feature combinations are determined again.</p></li>
<li><p>For each combination of features (<span class="math inline"><em>x</em></span>, <span class="math inline"><em>y</em></span>), a classifier is used to construct the boundary that separates the classes. In other words, a separate classifier is created for each combination, in an attempt to draw a line or a curve between classes.</p></li>
</ol>
<p><img src="./images/Figure-6-rbf-boundary.png" alt="Figure 6" /> <strong>Figure 6.</strong> Separation of two classes (red and blue) using radial SVM.</p>
<ol start="3" type="1">
<li><p>The classifier gives the probability of belonging of each unit to one class or another. The resulting probabilities become the weights of the edges between nodes(<span class="math inline"><em>x</em></span>, <span class="math inline"><em>y</em></span>). Thus, each experimental unit receives the weight of an edge between nodes (<span class="math inline"><em>x</em></span>, <span class="math inline"><em>y</em></span>).</p></li>
<li><p>Again, two principal approaches can take place - a network-weighted approach, where the weight of an edge is a calculated probability, or a thresholding approach, where an edge will be built if the probability is above a certain threshold.</p></li>
<li><p>As a result, all experimental units change their representation to weighted graphs.</p></li>
</ol>
<h4 id="correlation-graphs-and-networks">Correlation graphs and networks</h4>
<p>Correlation graphs are most often used when it is possible to obtain several consecutive measurements of the same features from the unit. If we have many such experimental units, we should be able to make the same number of such consecutive measurements, and the characteristics should be identical.</p>
<p>This concept is based on the hypothesis of Alexander Gorban <span class="citation" data-cites="Gorban2009May">[4]</span>, according to which in a certain system with many features under the condition of stability there are certain baseline correlations and variance <span class="math inline">(<em>C</em><em>o</em><em>r</em>,<em>V</em><em>a</em><em>r</em>)</span>. Over time random fluctuations in correlations and variance occur but they deviate slightly from the baseline.</p>
<p>Under the condition of a certain “stress” or influence on the system, the correlation of features drastically increases, as well as the variance <span class="math inline">(<em>C</em><em>o</em><em>r</em>↑,<em>V</em><em>a</em><em>r</em>↑)</span>.</p>
<p>In the future the system either returns to stable state again (in this case we will observe the return of correlation and variance to baseline <span class="math inline">(<em>C</em><em>o</em><em>r</em>↓,<em>V</em><em>a</em><em>r</em>↓)</span>) or complete disadaptation (or crisis) will come, at which correlations are significantly reduced (including those below the baseline level), while the variance remains large or increases even more <span class="math inline">(<em>C</em><em>o</em><em>r</em>↓,<em>V</em><em>a</em><em>r</em>↑)</span>.</p>
<p>The general steps for constructing a correlation graph are as follows:</p>
<ol type="1">
<li><p>We determine the features of the experimental unit that are changing over time, for example, <span class="math inline">(<em>A</em>,<em>B</em>,<em>C</em>)</span>.</p></li>
<li><p>Researchers choose an initial “time window,” for example, measurements at time <span class="math inline"><em>T</em></span> and some empirical size of this window. Suppose the window size is chosen to be 20 time points. Then we get 19 previous time points and time <span class="math inline"><em>T</em></span> for each feature measurements (i.e., 20 measurements in total):</p></li>
</ol>
<table>
<thead>
<tr class="header">
<th style="text-align: center;">Time point</th>
<th style="text-align: center;">A</th>
<th style="text-align: center;">b</th>
<th style="text-align: center;">c</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: center;"><span class="math inline"><em>T</em></span></td>
<td style="text-align: center;"><span class="math inline"><em>a</em><sub>0</sub></span></td>
<td style="text-align: center;"><span class="math inline"><em>b</em><sub>0</sub></span></td>
<td style="text-align: center;"><span class="math inline"><em>c</em><sub>0</sub></span></td>
</tr>
<tr class="even">
<td style="text-align: center;"><span class="math inline"><em>T</em><sub>−1</sub></span></td>
<td style="text-align: center;"><span class="math inline"><em>a</em><sub>−1</sub></span></td>
<td style="text-align: center;"><span class="math inline"><em>b</em><sub>−1</sub></span></td>
<td style="text-align: center;"><span class="math inline"><em>c</em><sub>−1</sub></span></td>
</tr>
<tr class="odd">
<td style="text-align: center;"><span class="math inline"><em>T</em><sub>−2</sub></span></td>
<td style="text-align: center;"><span class="math inline"><em>a</em><sub>−2</sub></span></td>
<td style="text-align: center;"><span class="math inline"><em>b</em><sub>−2</sub></span></td>
<td style="text-align: center;"><span class="math inline"><em>c</em><sub>−2</sub></span></td>
</tr>
<tr class="even">
<td style="text-align: center;">…</td>
<td style="text-align: center;">…</td>
<td style="text-align: center;">…</td>
<td style="text-align: center;">…</td>
</tr>
<tr class="odd">
<td style="text-align: center;"><span class="math inline"><em>T</em><sub>−19</sub></span></td>
<td style="text-align: center;"><span class="math inline"><em>a</em><sub>−19</sub></span></td>
<td style="text-align: center;"><span class="math inline"><em>b</em><sub>−19</sub></span></td>
<td style="text-align: center;"><span class="math inline"><em>c</em><sub>−19</sub></span></td>
</tr>
</tbody>
</table>
<ol type="1">
<li><p>The time window with features will be the basis for the future graph. The names of the features become the nodes of the graph. Next we calculate pairwise correlations between all the features. The correlation coefficient becomes the edge weight between the nodes.</p></li>
<li><p>The method of estimating the correlation is directly related to the types of underlying distributions of characteristics and hypotheses of researchers. In general, the Pearson, Spearman, Cramer’s V, Tau and others methods can be used to create correlation graphs.</p></li>
<li><p>It is worth paying attention to the significance of the correlation. In the absence of a significant correlation, it seems logical not to build an edge between the corresponding nodes, while the presence of a significant correlation allows you to build an edge and assign weight to it. This approach combines from both weighted graphs approach and the threshold approach (in this case the threshold is the level of significance).</p></li>
<li><p>Thus, we have obtained a correlation graph at time <span class="math inline"><em>T</em></span>.</p></li>
<li><p>The next stage is to “move” the whole window forward by 1 time period - to <span class="math inline"><em>T</em><sub>+1</sub></span>. After that the graph is built already for this timepoint. The order of manipulations is similar, the sequence of points 3). - 5). is repeated. Similar a new graph is created for the time <span class="math inline"><em>T</em><sub>+1</sub></span>, “time window” moves to the enxt point and so on.</p></li>
</ol>
<h3 id="further-manipulations-with-networks">Further manipulations with networks</h3>
<p>As we transformed the presentation of units into graphs, further manipulations are connected with analysing network structures of graphs.</p>
<p>The following approaches are found in the literature:</p>
<ul>
<li>calculation of descriptive statistics (minimum, maximum, mean, variance, coefficient of variation) of the main network characteristics closeness, betweenness, edge betweenness, page rank, eigen centrality, authority score, strength, edge weights;</li>
<li>calculation of the number of “0” values for each of these characteristics where appropriate;</li>
<li>with a reasonable number of graph vertices - raw degree values for each vertex in the weighted graph;</li>
<li>calculation of descriptive statistics (maximum, mean, variance) for shortest path lengths;</li>
<li>distances between the vertices of the weighted graph (maximum and minimum, average);</li>
<li>calculation of various matrix norms for the graph, e.g. <span class="math inline"><em>L</em><sub>1</sub></span>-norm;</li>
<li>calculation of the number of principal components using principal component analysis with fixed value of explained variance;</li>
<li>other indicators that will help to separate the graphs into classes;</li>
</ul>
<p>Another direction could be subgraph analysis, such as minimal spanning trees or weight-inverted minimal spanning trees.</p>
<p>The methods we discussed earlier imply that “unnormal” units will have more edges in graphs in case of threshold approach or larger edge weights in case of weighted approach. The key point is to determine the features that contribute most to the differences between “normal” and “unnormal” groups.</p>
<p>One of the approaches is a concept of weight-inverted minimal spanning tree. How to get such a tree? For each graph it is necessary to calculate the inverse weights of the edges, or <span class="math inline">$\frac{1}{w}$</span>, and then build a minimum spanning tree for such a graph, next recalculate the values of the weights back. Obviously this approach is most beneficial when constructing weighted graphs, or when using a combination of weighted and threshold approaches.</p>
<p>The weight-inverted minimal spanning tree should contain vertices connected by edges along which the given experimental unit differs most from the “normal” ones. This can help the researches to find the features that differ mostly from the “normal” ones. This approach takes into account the multidimensional relations between the features.</p>
<p><img src="./images/Figure-7-span-trees-weighted-full-graph.png" alt="Figure 7" /> <strong>Figure 7.</strong> a) Original weighted graph. b) Inverse-weighted graph. c) Minimal spanning tree for the original graph d) Weight-inverted minimal spanning tree.</p>
<p>For such trees, the above graph characteristics calculations are possible, but it is worth taking into consideration that the number of vertices for subgraphs may differ from the original graph and from other subgraphs.</p>
<h2 id="notes">Notes</h2>
<h2 class="unnumbered" id="references">References</h2>
<div id="refs" class="references csl-bib-body" role="doc-bibliography">
<div id="ref-Zanin2011Sep" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">1. </div><div class="csl-right-inline">Zanin M, Boccaletti S. <span class="nocase">Complex networks analysis of obstructive nephropathy data</span>. <em>Chaos: An Interdisciplinary Journal of Nonlinear Science</em>. 2011;21(3):033103. doi:<a href="https://doi.org/10.1063/1.3608126">10.1063/1.3608126</a></div>
</div>
<div id="ref-Nazarenko2021Oct" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">2. </div><div class="csl-right-inline">Nazarenko T, Whitwell HJ, Blyuss O, Zaikin A. <span class="nocase">Parenclitic and Synolytic Networks Revisited</span>. <em>Front Genet</em>. 2021;12. doi:<a href="https://doi.org/10.3389/fgene.2021.733783">10.3389/fgene.2021.733783</a></div>
</div>
<div id="ref-Krivonosov2022Jan" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">3. </div><div class="csl-right-inline">Krivonosov M, Nazarenko T, Bacalini MG, et al. <span class="nocase">Age-related trajectories of DNA methylation network markers: a parenclitic network approach to a family-based cohort of patients with Down Syndrome</span>. <em>bioRxiv</em>. Published online January 2022:2020.03.10.986505. <a href="https://doi.org/10.1101/2020.03.10.986505">https://doi.org/10.1101/2020.03.10.986505</a></div>
</div>
<div id="ref-Gorban2009May" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">4. </div><div class="csl-right-inline">Gorban AN, Smirnova EV, Tyukina TA. <span class="nocase">Correlations, Risk and Crisis: From Physiology to Finance</span>. <em>arXiv</em>. Published online May 2009. doi:<a href="https://doi.org/10.1016/j.physa.2010.03.035">10.1016/j.physa.2010.03.035</a></div>
</div>
<div id="ref-Whitwell2018Apr" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">5. </div><div class="csl-right-inline">Whitwell HJ, Blyuss O, Menon U, Timms JF, Zaikin A. <span class="nocase">Parenclitic networks for predicting ovarian cancer</span>. <em>Oncotarget</em>. 2018;9(32):22717-22726. doi:<a href="https://doi.org/10.18632/oncotarget.25216">10.18632/oncotarget.25216</a></div>
</div>
</div>
</body>
</html>
